### PR TITLE
Add optional oracle provider compilation

### PR DESCRIPTION
**Pull request by mhugo**
@elemoine What do you think ?

(There should be a way to avoid being root for alien ...)

### CODE REVIEW

**Comment by elemoine** [comment]
Sounds good! As discussed let's try to use fakeroot to avoid running the container as root.

**Comment by elemoine** [comment]
@mhugo have you made progress with this?

**Comment by mhugo** [comment]
PR updated that should fix root/non-root issues and compile the Oracle provider

**Comment by mhugo** [comment]
@elemoine

File: qgis-build/README.md [hunk 1]
```diff
@@ -106,3 +106,24 @@ Now to run QGIS Server with `gdb`:
 ```
 
 The `handle SIG33 nostop noprint pass` command is related to some issue in `gdb`. See https://bugzilla.redhat.com/show_bug.cgi?id=162402 for example.
+
+## Option: Oracle provider compilation
+
+If you want to generate .deb files for QGIS with support for the Oracle provider, you have to copy the following RPM files from Oracle into
```
Comment (elemoine) [inline qgis-build/README.md]: Do you know where these RPMs can be downloaded from?

File: qgis-build/README.md [hunk 2]
```diff
@@ -106,3 +106,24 @@ Now to run QGIS Server with `gdb`:
...
+your `build` directory created before:
+- oracle-instantclient12.1-basiclite-12.1.0.2.0-1.x86_64.rpm
+- oracle-instantclient12.1-devel-12.1.0.2.0-1.x86_64.rpm
+
+An environment variable `BUILD_ORACLE_PROVIDER` set to `y` will trigger the compilation and packaging:
```
Comment (elemoine) [inline qgis-build/README.md]: Do you want to add "of the Oracle provider" or something?

File: qgis-build/README.md [hunk 3]
```diff
@@ -106,3 +106,24 @@ Now to run QGIS Server with `gdb`:
...
+$ docker run -it --rm -v $(pwd):/qgis -v $HOME/.ccache:/.ccache -e USER=$(id -u):$(id -g) -e CCACHE_DIR=/.ccache -e BUILD_ORACLE_PROVIDER=y qgis-build
+```
+
+## Known issues
+
+Some utilities from Qt uses the `statx` syscall that needs the container to have some special privileges. This privileges may be present if your host has a `libseccomp` >= 2.3.3.
+
+See the discussion in the [corresponding Docker PR](https://github.com/moby/moby/pull/36417)
+
+For older hosts, a known workaround is to run docker commands in privileged mode by adding `--privileged` on the command line.
```
Comment (elemoine) [inline qgis-build/README.md]: "older hosts" ? What do you mean exactly?

File: qgis-build/build-debs.sh [hunk 4]
```diff
@@ -5,10 +5,22 @@
 
 set -e
 
+cd /qgis
+
+if [ "${BUILD_ORACLE_PROVIDER}" = "y" ]; then
```
Comment (elemoine) [inline qgis-build/build-debs.sh]: I've learned that double-blackets are preferable.

**Comment by elemoine** [comment]
Looks good. I added a commit with a reformulation in the README, and a couple of other comments.

**Comment by mhugo** [comment]
@elemoine Thanks. I've updated with modifications from your comments

**Comment by elemoine** [comment]
I added a additional commit replacing `[` with `[[` in `build-deb.sh`. Let's merge if you agree with my last commit. Also, I'd be in favor of squashing the three commits into just one commit before merging.

Thank you @mhugo.

**Comment by mhugo** [comment]
Merged. Thanks for your review @elemoine  !

**Pull request merged**